### PR TITLE
adding hasMultipleDimensions check to hasLegend clause

### LINKED ISSUE #4995: Legends disappear when there is only one row in the results.
### Bugs
If you're filing an issue about a bug please include as much information
as you can including the following.

- Your browser and the version: (e.x. Chrome 52.1, Firefox 48.0, IE 10)
Chrome

- Your operating system: (e.x. OS X 10, Windows XP, etc)
OS X 10.12.2

- Your databases: (e.x. MySQL, Postgres, MongoDB, …)
Redshift

- Metabase version: (e.x. 0.19.3)
0.24.0.RC2

- Metabase hosting environment: (e.x. Mac app, Elastic Beanstalk, Docker, Heroku, Linux/Ubuntu 12)
docker

- Metabase internal database: (e.x. H2 (default), MySQL, Postgres)
postgres

I choose two series for X and one series for Y, the legends are shown on the top left when there are many rows in results. And then I change the filter conditions to return only one row in results, the legends are gone, this sometimes may be annoying.

### After filter the results

Thanks for being part of the Metabase project!
-------

### PR DESCRIPTION
**Pull request by npfitz**
Closes https://github.com/metabase/metabase/issues/4995

### Description
We have logic that uses a few properties to determine whether or not a legend is shown in a line/area/bar chart. If the chart has multiple series, if we're able to change the series (dashcards), or it does not have a title but has actionButtons (also Dashcards... I think). However, if you had a multi series chart and hid or filtered all but one series, the legend would dissappear. This change adds a new check to the logic which is: does the graph have more than 1 dimension. Multiple dimensions implies that the series was intentionally broken out, and when this bug was filed, knowing that breakoutDimension the series on the screen was referencing may not have been as obvious (we now show it right on the data screen).

### How to verify
1. New -> Question -> Count of Products, grouped by Created At: Year, Category
2. Visualize, go to settings and hide all but one series. The legend should still persist
3. enable all series, go to filter and filter out all but one category, the legend should still persist

### Demo
image

### Checklist
- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: e2e/test/scenarios/visualizations/line_chart.cy.spec.js [hunk 1]
```diff
@@ -415,8 +415,13 @@ describe("scenarios > visualizations > line chart", () => {
...
     });
 
-    it.skip("should not drop the chart legend (metabase#4995)", () => {
+    it("should not drop the chart legend (metabase#4995)", () => {
       cy.findAllByTestId("legend-item").should("contain", "Doohickey");
+
+      cy.log("Ensure that legend is hidden when not dealing with multi series");
+      cy.findByTestId("viz-settings-button").click();
+      cy.findByTestId("remove-CATEGORY").click();
+      cy.get("main main").should("not.contain", "Doohickey");
```
Comment (EmmadUsmani) [inline e2e/test/scenarios/visualizations/line_chart.cy.spec.js:424]: > `cy.get("main main")`

Should we define a function in `e2e-ui-elements-helpers.js` to give this a more semantic name (e.g. `visualization()`)?

Comment (npfitz) [reply e2e/test/scenarios/visualizations/line_chart.cy.spec.js:424]: It's not a bad idea. I was actually surprised one didn't exist already lol. I could move it and name it something like `queryBuilderMain()`

**Review by EmmadUsmani — APPROVED** [review]

File: frontend/src/metabase/visualizations/components/LineAreaBarChart.jsx [hunk 2]
```diff
@@ -273,9 +273,14 @@ export default class LineAreaBarChart extends Component {
     const canSelectTitle = cardIds.size === 1 && onChangeCardAndRun;
 
     const hasMultipleSeries = series.length > 1;
+    const hasMultipleDimensions = settings["graph.dimensions"].length > 1;
```
Comment (alxnddr) [inline frontend/src/metabase/visualizations/components/LineAreaBarChart.jsx:276]: We usually call this `hasBreakout`

Comment (npfitz) [reply frontend/src/metabase/visualizations/components/LineAreaBarChart.jsx:276]: You'd think I'd know that by now 😂. I'll update it

**Review by alxnddr — APPROVED** [review]
Looks good! Please have a look at a couple of suggestions before merging

**Pull request merged**